src/App.tsx

```diff
@@ -1,61 +1,56 @@
-import React from 'react';
+import React, { useCallback, useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Autocomplete } from './Autocomplete';
+import { debounce } from 'lodash';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [debounceQuery, setDebounceQuery] = useState<string>('');
 
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+  useEffect(() => {
+    if (query !== '' && !selectedPerson) {
+      setSelectedPerson(null);
+    }
+  }, [query, selectedPerson])
 
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+  const filteredPeople = query.trim() === '' ? peopleFromServer : peopleFromServer.filter(person => person.name.toLowerCase().includes(debounceQuery.toLowerCase())
+  );
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+  const handleSelectPerson = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setDebounceQuery('');
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  const delay = 300;
```

Comment:
The debounce delay should be customizable via props with a default value of 300ms. Consider passing the delay as a prop to the component.